prov/util/src/util_mr_cache.c

```diff
@@ -121,9 +150,11 @@ void ofi_mr_cache_notify(struct ofi_mr_cache *cache, const void *addr, size_t le
 	iov.iov_base = (void *) addr;
 	iov.iov_len = len;
 
+	pthread_mutex_lock(&cache->stale_lock);
 	for (entry = cache->storage.overlap(&cache->storage, &iov); entry;
 	     entry = cache->storage.overlap(&cache->storage, &iov))
```

Comment:
The problem with this change is that the overlap find operation does not include the stale check as part of its search.

Reply:
Still not quite done with that yet. I think we can avoid the stale check if we prune the entry in the first place.